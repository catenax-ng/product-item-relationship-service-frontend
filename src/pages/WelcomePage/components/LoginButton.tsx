import { useKeycloak } from "@react-keycloak/web";
import { Button } from "cx-portal-shared-components";
import { useTranslation } from "react-i18next";
import { getCurrentEnvironment } from "../../../utils/sessionStorageHandling";

/**
 * The Login button is used to navigate to the keycloak logging page.
 * If the user is already logged in, the user will be logged out.
 * This is to ensure that the user is authenticated for the currently selected server environment.
 * @returns React.Component
 */
export const LoginButton: React.FC = () => {
  const { keycloak } = useKeycloak();
  const { t } = useTranslation();

  const clickHandler = async () => {
    if (keycloak.authenticated) {
      await keycloak.logout();
    }
    const serverEnv = getCurrentEnvironment();
    await keycloak.login({ redirectUri: `${window.location.origin}/${serverEnv}/dashboard` });
  };

  return (
    <Button onClick={clickHandler} style={{ marginTop: "75px" }}>
      {t("content.irs.welcomePage.loginButtonLabel")}
    </Button>
  );
};
