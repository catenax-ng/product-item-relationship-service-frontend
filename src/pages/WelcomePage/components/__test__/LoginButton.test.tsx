import { vi } from "vitest";
import { fireEvent, renderWithRouter, screen } from "../../../../testing/test-utils";
import { LoginButton } from "../LoginButton";
const login = vi.fn();

vi.mock("@react-keycloak/web", () => {
  return {
    useKeycloak: () => ({
      keycloak: {
        login: () => {
          login();
        },
      },
    }),
  };
});

it("clicks the login button", () => {
  const { container } = renderWithRouter(<LoginButton />);
  expect(container).toMatchSnapshot();

  const button = screen.getByRole("button", { name: /login/i });
  expect(button).toBeDefined();
  fireEvent.click(button);

  expect(login).toHaveBeenCalledTimes(1);
});
