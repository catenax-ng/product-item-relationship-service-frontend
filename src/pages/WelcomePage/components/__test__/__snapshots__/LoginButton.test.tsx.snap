// Vitest Snapshot v1

exports[`clicks the login button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    style="margin-top: 75px;"
    tabindex="0"
    type="button"
  >
    content.irs.welcomePage.loginButtonLabel
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
